### PR TITLE
feat: Support in RT Trainer for multiple entities.

### PR DESCRIPTION
**Pull request by NicolaiRee**
Solving #143 by expanding the Regression Transformer trainer to support multi-entity discriminations, i.e., support the multientity_cg collator from the RT repo.

Signed-off-by: Nicolai Ree <Ree@sunray>

### CODE REVIEW

File: src/gt4sd/algorithms/conditional_generation/regression_transformer/implementation.py [hunk 1]
```diff
@@ -90,9 +90,7 @@ def __init__(
             raise OSError(
                 f"algorithm_version {resources_path.split('/')[-1]} does not exist."
             )
-        self.tokenizer = InferenceBertTokenizer.from_pretrained(
-            resources_path, pad_even=False
-        )
```
Comment (jannisborn) [inline src/gt4sd/algorithms/conditional_generation/regression_transformer/implementation.py:95]: This was probably changed by mistake, i think we should keep the `pad_even=False` here

File: src/gt4sd/training_pipelines/regression_transformer/core.py [hunk 2]
```diff
@@ -85,6 +85,40 @@ class RegressionTransformerTrainingArguments(
...
+            "help": "The entity that is being masked during training. 0 corresponds to first entity and so on. -1 corresponds to "
+            "a random sampling scheme where the entity-to-be-masked is determined "
+            "at runtime in the collator. NOTE: If 'mask_entity_separator' is true, "
+            "this argument will not have any effect. Defaults to -1."
+        },
+    )
+    entity_separator_token: str = field(
+        default=".",
+        metadata={
+            "help": "The token that is used to separate "
```
Comment (jannisborn) [inline src/gt4sd/training_pipelines/regression_transformer/core.py:110]:
```suggestion
            "help": "Only applies if `cg_collator='multientity_cg'`.The token that is used to separate "
```

File: src/gt4sd/training_pipelines/regression_transformer/core.py [hunk 3]
```diff
@@ -85,6 +85,40 @@ class RegressionTransformerTrainingArguments(
...
+            "entities in the input. Defaults to '.' (applicable to SMILES & SELFIES)"
+        },
+    )
+    mask_entity_separator: bool = field(
+        default=False,
+        metadata={
+            "help": "Whether or not the entity separator token can be masked. If True, *all** textual tokens can be masked and we "
```
Comment (jannisborn) [inline src/gt4sd/training_pipelines/regression_transformer/core.py:117]:
```suggestion
            "help": "Only applies if `cg_collator='multientity_cg'`. Whether or not the entity separator token can be masked. If True, *all** textual tokens can be masked and we "
```

File: src/gt4sd/training_pipelines/regression_transformer/core.py [hunk 4]
```diff
@@ -85,6 +85,40 @@ class RegressionTransformerTrainingArguments(
...
+            "just default to the parent class behavior. If False, the exact behavior "
```
Comment (jannisborn) [inline src/gt4sd/training_pipelines/regression_transformer/core.py:118]:
```suggestion
            "the collator behaves like the `vanilla_cg ` even though it is a `multientity_cg`. If False, the exact behavior "
```

**Comment by jannisborn** [comment]
Hey @NicolaiRee,
Cool, the tests are passing now. Did you follow the steps suggested by the CLA bot to make the verification checks passing?

**Comment by NicolaiRee** [comment]
Hey @jannisborn, I have followed the direct instruction about the email and made the issue for getting on the contribution list, but I do not know why the verification still fails. Do you have any suggestions?

**Comment by drugilsberg** [comment]
> Hey @jannisborn, I have followed the direct instruction about the email and made the issue for getting on the contribution list, but I do not know why the verification still fails. Do you have any suggestions?

I think the problem is the first commit that is associated to an unverified account (see image).
image

At this point it's probably easy to:
- close this PR
- fork gt4sd:main again and apply all changes in a single commit with the verified account
- open another PR

**Comment by NicolaiRee** [comment]
Okay lets try that

**Pull request closed without merging**